apps/coscrad-frontend/src/utils/generic-components/qr-codes/qr-code-for-this-page.tsx

```diff
@@ -0,0 +1,20 @@
+import { Accordion, AccordionDetails, AccordionSummary } from '@mui/material';
+import QRCode from 'qrcode';
+import { useEffect, useRef } from 'react';
+
+export const QRCodeForThisPage = (): JSX.Element => {
+    const canvasRef = useRef();
+
+    useEffect(() => {
+        if (canvasRef.current) QRCode.toCanvas(canvasRef?.current, window.location.href);
```

Comment:
Is there ever a scenario where this condition is false? Maybe failed to generate over poor internet connection? Loading state?